lib/SIL/Parser/ParseSIL.cpp

```diff
@@ -1006,6 +1007,8 @@ static bool parseDeclSILOptional(bool *isTransparent,
       *isDynamic = IsDynamic;
     else if (isDistributed && SP.P.Tok.getText() == "distributed")
       *isDistributed = IsDistributed;
+    else if (isDistributed && SP.P.Tok.getText() == "ad_hoc_requirement_witness")
```

Comment:
Should check `isAdHocWitness` here

Reply:
by the way I don't really understand where those tokens we match against comr from here hmmm

Reply:
ok these come from SILPrinter